addons/l10n_pk_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,470 @@
+        if not re.match("^.{3,50}$", partner.state_id.name or ""):
+            message.append(_("- State required min 3 and max 50 characters"))
+        if message:
+            action = {
+                "view_mode": "form",
+                "type": "ir.actions.act_window",
+                "res_id": partner.id,
+                "res_model": "res.partner",
+                "views": [[self.env.ref("base.view_partner_form").id, "form"]],
+            }
+            error_msg = '\n'.join(map(str, message))
+            raise RedirectWarning(
+                _("Please enter required details for %s \n%s", partner.display_name, error_msg),
+                action,
+                _('Go to Configuration')
+            )
+
+    def _l10n_pk_validate_buyer(self, partner):
+        self.ensure_one()
+        message = []
+        if not re.match("^.{3,50}$", partner.state_id.name or ""):
+            message.append(_("- State required min 3 and max 50 characters"))
+        if message:
+            action = {
+                "view_mode": "form",
+                "type": "ir.actions.act_window",
+                "res_id": partner.id,
+                "res_model": "res.partner",
+                "views": [[self.env.ref("base.view_partner_form").id, "form"]],
+            }
+            error_msg = '\n'.join(map(str, message))
+            raise RedirectWarning(
+                _("Please enter required details for %s \n%s", partner.display_name, error_msg),
+                action,
+                _('Go to Configuration')
+            )
+
+    def format_timestamp(self, value):
+        return value.strftime('%Y-%m-%dT%H:%M:%S.%f')[:-3] + 'Z'
+
+    @api.model
+    def _l10n_pk_round_value(self, amount, precision_digits=2):
+        """
+            This method is call for rounding.
+            If anything is wrong with rounding then we quick fix in method
+        """
+        return round(amount, precision_digits) or 0.0  # avoid -0.0
+
+    def _l10n_pk_edi_extract_digits(self, string):
+        if not string:
+            return string
+        matches = re.findall(r"\d+", string)
+        result = "".join(matches)
+        return result
+
+    def _l10n_pk_is_global_discount(self, line):
+        return not line.tax_ids and line.price_subtotal < 0 or False
+
+    def _get_l10n_pk_edi_invoice_type(self, move_type):
+        match move_type:
+            case "in_invoice":
+                return 1
+            case "out_invoice":
+                return 2
+            case "in_refund":
+                return 3
+            case "out_refund":
+                return 4
+
+    def _get_l10n_pk_edi_fbr_ref_number(self, move):
+        if move.move_type in ['in_refund', 'out_refund']:
+            return move.reversed_entry_id._get_l10n_pk_edi_response_json().get("result")
+        return False
+
+    def _get_l10n_pk_edi_sro_serial_no(self, product_id):
+        return product_id.l10n_pk_edi_schedule_code.schedule_code[-3:]
```

Comment:
It's just used once and I don't think it's going to be overridden.
You can remove the function and inline the code.